### PR TITLE
send grpc with meta

### LINKED ISSUE #99: Pass grpc metadata
The python grpc code:
```python
import grpc
from your_grpc_module import your_pb2, your_pb2_grpc

# Create gRPC channel
channel = grpc.insecure_channel('localhost:50051')
stub = your_pb2_grpc.YourServiceStub(channel)

# Define the metadata
metadata = [
    ('header-key1', 'header-value1'),
    ('header-key2', 'header-value2'),
]

# RPC call within metadata
response = stub.YourRpcMethod(your_pb2.YourRequest(), metadata=metadata)
```

### PR DESCRIPTION
**Pull request by NKcqx**
Support send & recv grpc message with meta-data.
Meta-data can be in the form of dict like `{"key": "value"}`, or two-dimensional tuple (as the same form of grpc api) like `(("key", "value"),)`

This close #99

### CODE REVIEW

File: fed/api.py [hunk 1]
```diff
@@ -59,6 +48,7 @@ def init(
     cross_silo_messages_max_size_in_bytes: int = None,
     cross_silo_timeout_in_seconds: int = 60,
     enable_waiting_for_other_parties_ready: bool = False,
+    metadata: Dict = None,
```
Comment (jovany-wang) [inline fed/api.py:51]: We should add docstring for the new added field.

File: fed/_private/constants.py [hunk 2]
```diff
@@ -17,6 +17,8 @@
 
 KEY_OF_JOB_CONFIG = b"JOB_CONFIG"
 
+KEY_OF_METADATA = b"METADATA"
```
Comment (jovany-wang) [inline fed/_private/constants.py:20]: b"METADATA": is this too general to be distinguished?

File: fed/api.py [hunk 3]
```diff
@@ -173,29 +163,30 @@ def init(
     compatible_utils.kv.initialize()
 
     cluster_config = {
-        KEY_OF_CLUSTER_ADDRESSES: cluster,
-        KEY_OF_CURRENT_PARTY_NAME: party,
-        KEY_OF_TLS_CONFIG: tls_config,
-        KEY_OF_CROSS_SILO_SERIALIZING_ALLOWED_LIST: cross_silo_serializing_allowed_list,
-        KEY_OF_CROSS_SILO_MESSAGES_MAX_SIZE_IN_BYTES:
+        constants.KEY_OF_CLUSTER_ADDRESSES: cluster,
```
Comment (jovany-wang) [inline fed/api.py:166]: Gooood change!

File: fed/barriers.py [hunk 4]
```diff
@@ -201,8 +203,9 @@ def __init__(
         self._party = party
         self._tls_config = tls_config
         self.retry_policy = retry_policy
-        config = fed_config.get_cluster_config()
-        set_max_message_length(config.cross_silo_messages_max_size)
+        self.metadata = fed_config.get_job_config().meta_data
```
Comment (jovany-wang) [inline fed/barriers.py:206]:
```suggestion
        self._metadata = fed_config.get_job_config().meta_data
```
Maybe the private member should be used a prefix?

File: fed/config.py [hunk 5]
```diff
@@ -40,13 +40,19 @@ def cross_silo_messages_max_size(self):
 
 
 class JobConfig:
-    def __init__(self) -> None:
-        pass
+    def __init__(self, raw_bytes: bytes) -> None:
+        self._data = cloudpickle.loads(raw_bytes)
+
+    @property
+    def meta_data(self):
```
Comment (jovany-wang) [inline fed/config.py:47]:
```suggestion
    def metadata(self):
```
Is it conflict with the keeping word in python?

Comment (NKcqx) [reply fed/config.py:47]: Yes, accidentally missing this, thx for that

File: fed/_private/constants.py (hunk 2 above)
Comment (NKcqx) [reply fed/_private/constants.py]: INDEED

**Review by jovany-wang — APPROVED** [review]
It looks good now.

**Pull request merged**